From 1cf3ef469ee46524ca5656ae3666210cd297d4b9 Mon Sep 17 00:00:00 2001
From: Li Zhou <li.zhou@windriver.com>
Date: Tue, 12 Sep 2023 14:20:59 +0800
Subject: [PATCH] kernel-std: Add pkgs without abiname for image/headers

Add 2 packages linux-image-stx-amd64/linux-headers-stx-amd64
which don't have abiname in their names. They depend on packages with
abiname in names. Then we can use these 2 packages in anywhere
that involves image/headers packages (e.g. Build-Depends/yaml config
and so on). When the abiname is changed later in any kernel upgrading
we don't need change above places involved any more.

We don't use the linux-image-amd64/linux-headers-amd64 as Debian does
because they are built by linux-signed-amd64, and coupled with signed
kernel. We don't follow Debian's signing process so we create 2 new
packages which are coupled with unsigned image.

BTW, rename package "linux-keys-@version@" to "linux-keys" because the
"@version@" isn't necessary for this package. Then the version numbers
can be completely removed from the Build-Depends of OOT kernel
modules' codes.

Add the 2 new packages in init control file of "debian" folder, or else
a wrong build sequence ( OOT kernel modules are ahead of kernel ) will
be determined because it is deduced from init control files of
packages. The build error for kernel modules is like:
"unsat-dependency: linux-headers-stx-amd64:amd64"

Signed-off-by: Li Zhou <li.zhou@windriver.com>
---
 debian/control                              | 16 ++++++++++++++++
 debian/rules.real                           |  4 ++--
 debian/templates/control.extra.in           | 16 ++++++++++++++++
 debian/templates/control.tools-versioned.in |  2 +-
 4 files changed, 35 insertions(+), 3 deletions(-)

diff --git a/debian/control b/debian/control
index 11c2e77..255cd94 100644
--- a/debian/control
+++ b/debian/control
@@ -14606,3 +14606,19 @@ Description: Linux for multiprocessor 32-bit PA-RISC (dummy package)
  This is a dummy transitional package.  It can be safely removed.
 Build-Profiles: <!stage1 !pkg.linux.nokernel !pkg.linux.nometa>
 
+Package: linux-image-stx-amd64
+Architecture: amd64
+Depends: linux-image-5.10.0-6-amd64-unsigned (= ${binary:Version}), ${misc:Depends}
+Description: Linux for 64-bit PCs (meta-package)
+ This package depends on the latest unsigned Linux kernel and modules for
+ use on PCs with AMD64, Intel 64 or VIA Nano processors.
+Build-Profiles: <!stage1 !pkg.linux.nokernel>
+
+Package: linux-headers-stx-amd64
+Architecture: amd64
+Depends: linux-headers-5.10.0-6-amd64 (= ${binary:Version}), ${misc:Depends}
+Description: Header files for Linux amd64 configuration (meta-package)
+ This package depends on the architecture-specific header files for the
+ latest Linux kernel amd64 configuration.
+Build-Profiles: <!stage1 !pkg.linux.nokernel>
+
diff --git a/debian/rules.real b/debian/rules.real
index a13e1f5..5a1fcee 100644
--- a/debian/rules.real
+++ b/debian/rules.real
@@ -72,7 +72,7 @@ binary-arch-arch: \
 	$(call if_package, $(SOURCE_BASENAME)-config-$(VERSION), install-config_$(ARCH)) \
 	$(call if_package, linux-libc-dev,               install-libc-dev_$(ARCH)) \
 	$(call if_package, linux-kbuild-$(VERSION),      install-kbuild) \
-	$(call if_package, linux-keys-$(VERSION),        install-keys) \
+	$(call if_package, linux-keys,                   install-keys) \
 	$(call if_package, linux-perf-$(VERSION),        install-perf) \
 	$(if $(filter-out x-$(VERSION),x$(SOURCE_SUFFIX)), \
 	  $(call if_package, linux-perf$(SOURCE_SUFFIX), install-perf-meta)) \
@@ -679,7 +679,7 @@ endif
 	dh_md5sums
 	dh_builddeb
 
-install-keys: PACKAGE_NAME = linux-keys-$(VERSION)
+install-keys: PACKAGE_NAME = linux-keys
 install-keys: DH_OPTIONS = -p$(PACKAGE_NAME)
 install-keys: DIR = $(CURDIR)/debian/$(PACKAGE_NAME)
 install-keys:
diff --git a/debian/templates/control.extra.in b/debian/templates/control.extra.in
index ef90151..c84ab8a 100644
--- a/debian/templates/control.extra.in
+++ b/debian/templates/control.extra.in
@@ -42,3 +42,19 @@ Priority: extra
 Depends: linux-image-parisc, ${misc:Depends}
 Description: Linux for multiprocessor 32-bit PA-RISC (dummy package)
  This is a dummy transitional package.  It can be safely removed.
+
+Package: linux-image-stx-amd64
+Build-Profiles: <!stage1 !pkg.linux.nokernel>
+Depends: linux-image-@abiname@-amd64-unsigned (= ${binary:Version}), ${misc:Depends}
+Architecture: amd64
+Description: Linux for 64-bit PCs (meta-package)
+ This package depends on the latest unsigned Linux kernel and modules
+ for use on PCs with AMD64, Intel 64 or VIA Nano processors.
+
+Package: linux-headers-stx-amd64
+Build-Profiles: <!stage1 !pkg.linux.nokernel>
+Depends: linux-headers-@abiname@-amd64 (= ${binary:Version}), ${misc:Depends}
+Architecture: amd64
+Description: Header files for Linux amd64 configuration (meta-package)
+ This package depends on the architecture-specific header files for the latest
+ Linux kernel amd64 configuration.
diff --git a/debian/templates/control.tools-versioned.in b/debian/templates/control.tools-versioned.in
index c9d48ed..9daecd7 100644
--- a/debian/templates/control.tools-versioned.in
+++ b/debian/templates/control.tools-versioned.in
@@ -6,7 +6,7 @@ Multi-Arch: foreign
 Description: Kbuild infrastructure for Linux @version@
  This package provides the kbuild infrastructure for the headers packages for Linux kernel version @version@.
 
-Package: linux-keys-@version@
+Package: linux-keys
 Build-Profiles: <!stage1 !pkg.linux.notools>
 Architecture: linux-any
 Depends: ${shlibs:Depends}, ${misc:Depends}
-- 
2.17.1

